crates/iceberg-catalog/src/catalog/io.rs

```diff
@@ -66,18 +67,34 @@ pub(crate) async fn read_file(file_io: &FileIO, file: &Location) -> Result<Vec<u
         file.to_string()
     };
 
-    retry_fn(|| async {
+    let content: Vec<_> = retry_fn(|| async {
         // InputFile isn't clone hence it's here
         file_io
             .clone()
             .new_input(file.clone())
-            .map_err(IoError::FileCreation)?
+            .map_err(IoError::FileInput)?
             .read()
             .await
             .map_err(|e| IoError::FileRead(Box::new(e)))
             .map(Into::into)
     })
-    .await
+    .await?;
+
+    if file.as_str().ends_with(".gz.metadata.json") {
+        let codec = CompressionCodec::Gzip;
+        let content = codec.decompress(&content)?;
```

Comment:
spawn_blocking?

Reply:
`write_metadata_file` should then also spawn_blocking right?